packages/amplify_datastore/example/lib/models/Blog.dart

```diff
@@ -53,15 +55,33 @@ class Blog extends Model {
     return _posts;
   }
 
-  const Blog._internal({required this.id, required name, posts})
-      : _name = name,
-        _posts = posts;
+  TemporalDateTime? get createdAt {
+    return _createdAt;
+  }
+
+  TemporalDateTime? get updatedAt {
+    return _updatedAt;
+  }
 
-  factory Blog({String? id, required String name, List<Post>? posts}) {
+  const Blog._internal(
+      {required this.id, required name, posts, createdAt, updatedAt})
+      : _name = name,
+        _posts = posts,
+        _createdAt = createdAt,
+        _updatedAt = updatedAt;
+
+  factory Blog(
+      {String? id,
+      required String name,
+      List<Post>? posts,
+      TemporalDateTime? createdAt,
+      TemporalDateTime? updatedAt}) {
```

Comment:
A general question: are `createdAt` and `updatedAt` managed by DynamoDB? If this is true, should we expose them as Model fields and allow manual modification?

Reply:
Hi @HuiSF wanted to make sure I understand what you're saying here.  So you're saying we shouldn't have createdAt and updatedAt as parameters in the factory Blog constructor method here because the user technically cannot modify them as they are automatically populated by AppSync?

I think we could make the following change - what are your thoughts?

```
  factory Blog(
      {String? id,
      required String name,
      List<Post>? posts}) {
    return Blog._internal(
        id: id == null ? UUID.getUUID() : id,
        name: name,
        posts: posts != null ? List<Post>.unmodifiable(posts) : posts,
        createdAt: null,
        updatedAt: null);
  }
```

Reply:
> because the user technically cannot modify them as they are automatically populated by AppSync?

Yes this is what I meant. These two fields should not be assignable/modifiable via constructor (including internal constructor). For Flutter use case, I think they are only assignable in `Model.fromJson` method (which is technically used internally by Flutter plugin to deserialize model data)

So I think these two fields should not be listed in the `_internal` constructor as well...

Reply:
Alright I'll update it like that.  We'll need to talk to codegen team to include those changes as well.

Reply:
We should wait on this PR until Aaron confirms he adds those codegen changes in then.  I'll contact him now about that.  Should be quick to make if he has the bandwidth now.

Reply:
Removing the createdAt and updatedAt fields from the constructor will cause issue with our existing copyWith implementation. 

```
  Blog copyWith(
      {String? id,
      String? name,
      List<Post>? posts,
      TemporalDateTime? createdAt,
      TemporalDateTime? updatedAt}) {
    return Blog(
        id: id ?? this.id,
        name: name ?? this.name,
        posts: posts ?? this.posts,
        createdAt: createdAt ?? this.createdAt,
        updatedAt: updatedAt ?? this.updatedAt);
  }
```

We could switch to using the internal constructor instead so that the public constructor doesn't have createdAt and updatedAt.  However we would need to keep internal constructor to have the createdAt and updatedAt timestamps as parameters then.